composables/useTimeSeriesGroups.ts

```diff
@@ -0,0 +1,24 @@
+import type { DisplayInfo, TimeSeriesGroup } from "~/types/apiResponseTypes";
+
+// Given the metadata for a time series *group* (that is, a collection of related time series, likely two)
+// and a boolean indicating whether to use daily or cumulative ('total') data,
+// return the active role (daily/total) and the metadata for the corresponding time series itself.
+export default (seriesGroup: MaybeRefOrGetter<TimeSeriesGroup>, isDaily: MaybeRefOrGetter<boolean>) => {
+  const appStore = useAppStore();
+
+  const activeRole = computed(() => {
+    // Since we are using a switch UI, we assume there are only two roles, and "total" role precedes "daily" role
+    const timeSeriesGroupRoles = Object.keys(toValue(seriesGroup).time_series).slice(0, 2); // ["total", "daily"]
+    return timeSeriesGroupRoles[Number(toValue(isDaily))];
```

Comment:
This feels like quite a convoluted way to return "total" or "daily" and feels a bit dangerous to be making assumptions about the object keys order. Would it be any more presumptuous just to do:
```suggestion
   return toValue(isDaily) ? "daily" : "total";
```
.. could potentially also do a bit of validation somewhere that the keys in the actual data are as expected..